core/Plugin/Controller.php

```diff
@@ -576,11 +583,16 @@ protected function setMaxDateView(Date $maxDate, $view)
      * Will exit on error.
      *
      * @param View $view
+     * @param string|null $viewType 'basic' or 'admin'. If null, set based on the type of controller.
```

Comment:
if someone extends this controller, and passes `admin` it wouldn't work as expected right?

Reply:
Are you suggesting if admin is passed, we should only allow it if `$this instanceof ControllerAdmin`? Or something else?

Reply:
I just meant might need to be updated in the docs ? 
The check with instanceof might be useful though and prevent false expectations if admin is passed. This way a developer would directly notice to extend different class.

Reply:
It would sort of work as expected if `'admin'` is passed, since `setBasicVariablesView()` is overridden in ControllerAdmin. I think there would be many more variables set (and possibly an exception if there's no idsite or something like that).

We could simply keep it undocumented except for `renderTemplate()`, doesn't really need to be set anywhere else I think.

Reply:
Or should we just move `setBasicVariablesNoneAdminView` etc to `plugin\Controller` and make it work even when it extends only `Plugin\Controller` and not `ControllerAdmin`? just a random thought, no preference in any direction here

Reply:
I'm not sure what that would solve?

Reply:
You have 

```php
class Foo extends Plugin\Controller {
   public function myAction(){
      return $this->renderTemplate('bar.twig', 'admin');
   }
}
``` 
would then maybe work instead of needing to change the parent controller to `class Foo extends Plugin\ControllerAdmin`? I haven't looked too much into the code though

Reply:
The problem is with:

```
class SitesManagerController extends Plugin\AdminController {
    public function myNormalAction() {
        return $this->renderTemplate('bar.twig');
    }
}
```

Where `ControllerAdmin::setBasicVariablesView()` calls `ControllerAdmin::setBasicVariablesAdminView()`, when we don't want it to.

Reply:
Or are you just talking about that specific case? I'm not sure if we should allow it, ie, should normal controllers be allowed to render admin pages?

Reply:
> should normal controllers be allowed to render admin pages?
No preference. It's just that `renderTemplate()` in `Plugin\Controller` has this parameter and it's documented to set `admin` optionally. So I would expect it to work even when `Plugin\Controller` is extended. We can also throw an exception in this case though.

Reply:
Will throw an exception. We could change `$viewType` to `$isBasicView` or something, but then a dev could supply `false`.

This isn't an ideal solution, but I'm not too sure of another easy/quick one.

Reply:
> Will throw an exception. 

👍